reconciler/reconciler/doctype/cd_gstr_2b_data_upload_tool/cd_gstr_2b_data_upload_tool.py

```diff
@@ -283,4 +304,276 @@ def get_supplier_by_gstin(gstin):
+						filters=[['company_gstin' ,'=',company_gstin],
+						['posting_date' ,'>=',from_date],
+						['docstatus', '=', 1],
+						['posting_date' ,'<=',to_date]],
+						fields=['name','supplier_gstin as gstin',
+						'bill_date as document_date',
+						'bill_no as document_number',
+						'total as total_taxable_amount'])
+	for row in pi_doc_list:
+		is_linked = frappe.db.get_value('CD GSTR 2B Entry', {'cf_purchase_invoice': row['name'], 'cf_match_status': 'Exact Match'},'name')
+		if not is_linked:
+			row['document_type'] = 'Invoice'
+			pr_list.append(row.update(get_tax_details(row['name'])))
+	return pr_list
+
+def get_tax_details(doc_name):
+	tax_amount = []
+	tax_details = {
+	'igst_amount':0,
+	'cgst_amount':0,
+	'sgst_amount':0,
+	'cess_amount':0,
+	'total_tax_amount': 0}
+
+	account_head_fields = ['igst_account','cgst_account','sgst_account','cess_account']
+	doc = frappe.get_doc('Purchase Invoice', doc_name)
+	gst_accounts = get_gst_accounts(doc.company)
+	for row in doc.taxes:
+		for accounts in gst_accounts.values():
+			if row.account_head in accounts:
+				if not type(accounts[-1]) == int:
+					accounts.append(0)
+				accounts[-1]+=row.tax_amount
+	for idx in range(len(account_head_fields)):
+		tax_amt = 0
+		if gst_accounts[account_head_fields[idx]][-1]  and not type(gst_accounts[account_head_fields[idx]][-1]) == str:
+			tax_amt = gst_accounts[account_head_fields[idx]][-1]
+		tax_details[list(tax_details.keys())[idx]] = tax_amt
+		tax_amount.append(tax_amt)
+	tax_details['total_tax_amount'] = sum(tax_amount)
+	return tax_details
+
+def last_day_of_month(year, month):
+	last_days = [31, 30, 29, 28]
+	for day in last_days:
+		try:
+			end = datetime(year, month, day)
+		except ValueError:
+			continue
+		else:
+			return end.date()
+	return None
+
+def get_match_status(gstr2b_doc, pr_list, amount_threshold = 1):
+	amount_params = {
+	'total_taxable_amount',
+	'total_tax_amount',
+	'igst_amount',
+	'cgst_amount',
+	'sgst_amount',
+	'cess_amount'}
+	gstin_matched_pr_list = []
+	gstin_doctype_matched_list = []
+	remaining_list = []
+	partial_match_list = []
+	probable_match_list = []
+	mismatch_list = []
+	for pr in pr_list:
+		if pr['gstin'] == gstr2b_doc['gstin']:
+			if not pr['document_type'] == gstr2b_doc['document_type']:
+				gstin_matched_pr_list.append(pr)
+			else:
+				gstin_doctype_matched_list.append(pr)
+		else:
+			remaining_list.append(pr)
+
+	for pr in gstin_doctype_matched_list:
+		reason = []
+		count = 0
+		if pr['document_date'] == gstr2b_doc['document_date']:
+			count += 1
+		else:
+			reason.append('Document Date')
+		if pr['document_number'] == gstr2b_doc['document_number']:
+			count += 1
+		else:
+			reason.append('Document Number')
+		for param in amount_params:
+			if (abs(pr[param] - gstr2b_doc[param])<amount_threshold):
+				count+=1
+			else:
+				reason.append(param.replace('_',' ').title())
+		if count == 8:
+			update_match_status(gstr2b_doc, [pr, 'Exact Match', reason])
```

Comment:
please don't "update" inside this "get" function

Reply:
Removed via https://github.com/aerele/reconciler/pull/4/commits/4bf29f674100887585696a22662c0ddccd5f5129. Kindly check this!